### PR TITLE
add some os stdlib functions

### PR DESCRIPTION
**Pull request by maybe-rachel**
Specifically: 
- os.pipe
- os.pipe2
- os.get_blocking
- os.set_blocking
- os.get_inheritable
- os.set_inheritable

Sorry for adding so many on my first PR. I was just planning on doing pipe at first, and kinda got dragged into the rest for testing.

Regarding os.pipe (~line 1000), is it okay to ignore close(fd) failure? That's what CPython does, and I'm not otherwise sure how to appropriately handle that. 

Also, flake8 complains about stdlib_os.py a lot. Not my code, and I didn't want to run a linter and make my first PR seem bigger.

### CODE REVIEW

File: tests/snippets/stdlib_os.py [hunk 1]
```diff
@@ -251,6 +251,69 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 	assert os.path.basename(fname) == FILE_NAME
 	assert os.path.dirname(fname) == tmpdir
 
+	# os.get_blocking, os.set_blocking
+	# TODO: windows support should be added for below functions
+	# os.pipe, 
+	# os.set_inheritable, os.get_inheritable, 
+	if os.name != "nt":
+		rfd, wfd = os.pipe()
```
Comment (coolreader18) [inline tests/snippets/stdlib_os.py]: I don't think these tests have to be in this `with` block, as they don't use `tmpdir` for anything

File: vm/src/stdlib/os.rs [hunk 2]
```diff
@@ -925,6 +927,102 @@ fn os_chdir(path: PyStringRef, vm: &VirtualMachine) -> PyResult<()> {
...
+    };
+    _set_flag().or_else(|err| Err(convert_nix_error(vm, err)))
+}
+
+#[cfg(unix)]
+fn os_pipe(vm: &VirtualMachine) -> PyResult<(RawFd, RawFd)> {
+    use nix::unistd::close;
+    use nix::unistd::pipe;
+    match pipe() {
+        Err(err) => Err(convert_nix_error(vm, err)),
```
Comment (coolreader18) [inline vm/src/stdlib/os.rs]: I think you could do something like `let (rfd, wfd) = pipe().map_err(|err| ...)?;`

File: vm/src/stdlib/os.rs [hunk 3]
```diff
@@ -925,6 +927,102 @@ fn os_chdir(path: PyStringRef, vm: &VirtualMachine) -> PyResult<()> {
...
+        Ok(ok) => {
+            os_set_inheritable(ok.0, false, vm)
+                .and_then(|_| os_set_inheritable(ok.1, false, vm))
+                .or_else(|err| {
+                    close(ok.0)
```
Comment (coolreader18) [inline vm/src/stdlib/os.rs]: You could do `let _ = close(...);` to ignore the result. I think ignoring the result is fine; Rust does the same thing when it calls `drop()` on a `std::fs::File`.

**Comment by maybe-rachel** [comment]
> Thank you for contributing @maybe-rachel! I've left a few comments, but overall this looks very good.

Thanks for all your suggestions! I did all those. Also, I converted all tabs to spaces because tabs were breaking it, so the diff's a bit bigger than it should be :X

**Comment by coolreader18** [comment]
Would you be able to find a way to not change all the indentation? It would just be nice for some git tools like blame.

**Comment by coolreader18** [comment]
Thanks again for contributing!

**Pull request merged**